### PR TITLE
feat(UI): Update dashboard rights

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This updates the lis of user roles

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Dashboards/Shares/translatedLabels.tsx [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export const labelDashboardAccessRights = 'Dashboard access rights';
+export const labelUserRolesIsUpdated = 'The user roles is updated';
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/Shares/translatedLabels.tsx:2]:
```suggestion
export const labelUserRolesAreUpdated = 'User roles are updated';
```

File: centreon/www/front_src/src/Dashboards/Shares/useShareUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+} from '@centreon/ui';
+
+import { getDashboardSharesEndpoint } from '../api/endpoints';
+import {
+  DashboardShare,
+  DashboardShareForm,
+  DashboardShareToAPI
+} from '../models';
+
+import { labelUserRolesIsUpdated } from './translatedLabels';
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/Shares/useShareUpdate.ts:20]:
```suggestion
import { labelUserRolesAreUpdated } from './translatedLabels';
```

File: centreon/www/front_src/src/Dashboards/Shares/useShareUpdate.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        previousSharesListing
+      };
+    }
+  });
+
+  const updateShares = (newShares: Array<DashboardShareForm>): void => {
+    const formattedShares = formatSharesToAPI(newShares);
+
+    mutateAsync(formattedShares).then(() => {
+      showSuccessMessage(t(labelUserRolesIsUpdated));
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/Shares/useShareUpdate.ts:82]:
```suggestion
      showSuccessMessage(t(labelUserRolesAreUpdated));
```

File: centreon/www/front_src/src/Dashboards/Dashboard/Dashboard.cypress.spec.tsx [hunk 4]
```diff
@@ -406,5 +419,21 @@ describe('Dashboard', () => {
...
+      cy.findByLabelText(labelShareTheDashboard).click();
+
+      cy.findAllByTestId('change_role').eq(0).parent().click();
+      cy.get('[data-value="editor"]').click();
+
+      cy.findByLabelText(labelSave).click();
+
+      cy.waitForRequest('@putDashboardShares');
+
+      cy.contains(labelUserRolesIsUpdated).should('be.visible');
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/Dashboard/Dashboard.cypress.spec.tsx:436]:
```suggestion
      cy.contains(labelUserRolesAreUpdated).should('be.visible');
```

File: centreon/www/front_src/src/Dashboards/Dashboard/Dashboard.cypress.spec.tsx [hunk 5]
```diff
@@ -19,14 +19,16 @@ import {
 } from '../api/endpoints';
 import { DashboardRole } from '../models';
 import { labelShareTheDashboard } from '../translatedLabels';
+import { labelUserRolesIsUpdated } from '../Shares/translatedLabels';
```
Comment (Thebarda) [inline centreon/www/front_src/src/Dashboards/Dashboard/Dashboard.cypress.spec.tsx:22]:
```suggestion
import { labelUserRolesAreUpdated } from '../Shares/translatedLabels';
```

**Comment by g10** [comment]
fyi. merging this sub-feature branch in the main feature branch and have the code review there

**Pull request merged**